main.py

```diff
@@ -14,22 +16,35 @@ def root():
     return 'Welcome to our CS467 capstone project! This serves as the backend (https://github.com/youncher/mathgo-backend) to our project, Math Go (https://github.com/youncher/mathgo).'
 
 @app.route('/user/validation', methods=['POST'])
-def user_validation():
-	# Check if user is an existing Math Go user
+def handle_validation():
+	# TODO: Update to accept https requests
+	# Verify user token; Check if existing Math Go user
 	if request.method == 'POST':
 		content = request.get_json()
+		provided_gid = content.get("gid")
+		provided_id_token = content.get("idToken")
 		
-		# TODO: Handle case if request body does not contain gid
+		# Missing required fields = sign in fail
```

Comment:
I see that you're returning  loginSuccessful here as a field from the backend, but I think it's not being attached inside the app?